core/reader/LogFileReader.cpp

```diff
@@ -2309,36 +2336,55 @@ LineInfo DockerJsonFileParser::NewGetLastLine(StringView buffer,
                                               bool needSingleLine,
                                               std::vector<BaseLineParse*>* lineParsers) {
     if (end == 0) {
-        return {.data = StringView(), .lineBegin = 0, .lineEnd = 0, .rollbackLineFeedCount = 0, .fullLine = false};
+        return {.data = StringView(),
+                .lineBegin = 0,
+                .lineEnd = 0,
+                .rollbackLineFeedCount = 0,
+                .fullLine = false,
+                .forceRollbackLineFeedCount = 0};
     }
     if (protocolFunctionIndex == 0) {
         // 异常情况, DockerJsonFileParse不允许在最后一个解析器
-        return {.data = StringView(), .lineBegin = 0, .lineEnd = 0, .rollbackLineFeedCount = 0, .fullLine = false};
+        return {.data = StringView(),
+                .lineBegin = 0,
+                .lineEnd = 0,
+                .rollbackLineFeedCount = 0,
+                .fullLine = false,
+                .forceRollbackLineFeedCount = 0};
     }
 
     size_t nextProtocolFunctionIndex = protocolFunctionIndex - 1;
     LineInfo finalLine;
     while (!finalLine.fullLine) {
         LineInfo rawLine = (*lineParsers)[nextProtocolFunctionIndex]->NewGetLastLine(
             buffer, end, nextProtocolFunctionIndex, needSingleLine, lineParsers);
-        if (rawLine.data.back() == '\n') {
+        if (rawLine.data.size() > 0 && rawLine.data.back() == '\n') {
             rawLine.data = StringView(rawLine.data.data(), rawLine.data.size() - 1);
         }
 
         LineInfo line;
         parseLine(rawLine, line);
-        finalLine.data = line.data;
-        finalLine.fullLine = line.fullLine;
-        finalLine.lineBegin = line.lineBegin;
-        finalLine.rollbackLineFeedCount += line.rollbackLineFeedCount;
-        finalLine.dataRaw = line.dataRaw;
-        if (finalLine.lineEnd == 0) {
-            finalLine.lineEnd = line.lineEnd;
+        int32_t rollbackLineFeedCount = 0;
+        int32_t forceRollbackLineFeedCount = 0;
+        if (line.fullLine) {
+            rollbackLineFeedCount = line.rollbackLineFeedCount;
+            forceRollbackLineFeedCount = finalLine.forceRollbackLineFeedCount;
+        } else {
```

Comment:
这里再看一下，有没有遗漏原来的forceRollbackLine
```suggestion
            line.forceRollbackLineFeedCount = finalLine.forceRollbackLineFeedCount;
```